src/state/slices/actionSlice/selectors.ts

```diff
@@ -0,0 +1,81 @@
+import { selectEnabledWalletAccountIds } from '../common-selectors'
+import { actionCenterSlice } from './actionSlice'
+import {
+  ActionStatus,
+  isLimitOrderPayloadDiscriminator,
+  isTradePayloadDiscriminator,
+} from './types'
+
+import { createDeepEqualOutputSelector } from '@/state/selector-utils'
+import { selectSwapIdParamFromFilter } from '@/state/selectors'
+
+export const selectActions = actionCenterSlice.selectors.selectActions
+
+export const selectInitializedActionsByUpdatedAtDescFilteredByWallet =
+  createDeepEqualOutputSelector(
+    selectActions,
+    selectEnabledWalletAccountIds,
+    (actions, enabledWalletAccountIds) => {
+      return [...actions]
+        .filter(
+          action =>
+            (action.initiatorAccountId &&
+              enabledWalletAccountIds.includes(action.initiatorAccountId)) ||
+            !action.initiatorAccountId,
+        )
+        .sort((a, b) => b.updatedAt - a.updatedAt)
+    },
+  )
+
+export const selectPendingActionsFilteredByWallet = createDeepEqualOutputSelector(
+  selectActions,
+  selectEnabledWalletAccountIds,
+  (actions, enabledWalletAccountIds) => {
+    return actions.filter(
+      action =>
+        action.status === ActionStatus.Pending &&
+        ((action.initiatorAccountId &&
+          enabledWalletAccountIds.includes(action.initiatorAccountId)) ||
+          !action.initiatorAccountId),
+    )
+  },
+)
+
+export const selectPendingSwapActionsFilteredByWallet = createDeepEqualOutputSelector(
+  selectActions,
+  selectEnabledWalletAccountIds,
+  (actions, enabledWalletAccountIds) => {
+    return actions.filter(
+      action =>
+        (action.status === ActionStatus.Pending || action.status === ActionStatus.Open) &&
+        ((action.initiatorAccountId &&
+          enabledWalletAccountIds.includes(action.initiatorAccountId)) ||
+          !action.initiatorAccountId),
+    )
+  },
+)
+
+export const selectOpenLimitOrderActionsFilteredByWallet = createDeepEqualOutputSelector(
+  selectActions,
+  selectEnabledWalletAccountIds,
+  (actions, enabledWalletAccountIds) => {
+    return actions.filter(
+      action =>
+        action.status === ActionStatus.Open &&
```

Comment:
q: how will we handle partial fills? Will this also be a status?

Reply:
I think it should remain "Open" but filled with `X%`

I don't think it's needed for a specific status, we can always select by filled amount !== 0